stl/inc/xutility

```diff
@@ -6202,28 +6202,22 @@ namespace ranges {
                 _Ptr_t _Result;
 #if _USE_STD_VECTOR_ALGORITHMS
                 if constexpr (_Is_sized) {
-                    // When _Is_sized && _Elements_are_1_or_2_bytes, prefer this over memchr()/wmemchr() for performance
+                    // When _Is_sized && _Elements_are_1_byte, prefer this over memchr() for performance
                     const auto _Last_ptr = _First_ptr + (_Last - _First);
                     _Result              = _STD _Find_vectorized(_First_ptr, _Last_ptr, _Val);
                 } else
 #endif // ^^^ _USE_STD_VECTOR_ALGORITHMS ^^^
                 {
-                    _STL_INTERNAL_STATIC_ASSERT(_Elements_are_1_or_2_bytes);
+                    _STL_INTERNAL_STATIC_ASSERT(_Elements_are_1_byte);
                     size_t _Count;
                     if constexpr (_Is_sized) {
                         _Count = static_cast<size_t>(_Last - _First);
                     } else {
                         _Count = SIZE_MAX;
                     }
 
-                    if constexpr (sizeof(_Iter_value_t<_It>) == 1) {
-                        _Result =
-                            static_cast<_Ptr_t>(_CSTD memchr(_First_ptr, static_cast<unsigned char>(_Val), _Count));
-                    } else {
-                        _STL_INTERNAL_STATIC_ASSERT(sizeof(_Iter_value_t<_It>) == 2);
-                        _Result = reinterpret_cast<_Ptr_t>(const_cast<wchar_t*>(_CSTD wmemchr(
-                            reinterpret_cast<const wchar_t*>(_First_ptr), static_cast<wchar_t>(_Val), _Count)));
-                    }
+                    // C23 7.32.4.6.9 "The wmemchr generic function"/2 doesn't work with unreachable_sentinel_t.

```

Comment:
This seems too indirect to me. Maybe link `memchr` behavior quote instead?
```c
// C23 7.27.5.2 "The memchr generic function"/2 as if reads the characters sequentially.
// wmemchr does not have to do that

Reply:
The same incorrect assumption about `wmemchr` is also made in `vector_algorithms.cpp`, it is in a compat path for no longer existent vectorization.

Reply:
I can update the comment and the `vector_algorithms.cpp` usage is super easy to fix - thanks, I totally missed that!